src/components/day.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react';
+import Cell from './cell';
+
+function Day({ row, mode, onChange }) {
+    const [range, setRange] = useState({
+        start: null,
+        end: null
+    });
+    const [mouseStatus, setMouseStatus] = useState({
+        down: false
+    });
+    const [grid, setGrid] = useState(row);
```

Comment:
The problem with clear not working is because of this duplicated `grid` state.
We can only have one grid state. (The one at the App). This one cannot exist, otherwise you'll have multiple sources of states.

You'll notice that even when you call the clear and set a new `row` on App, in this component you have a useEffect that runs when `grid` changes. `grid` is not the same as `row`

```
  useEffect(() => {
    onChange(grid);
  }, [grid]);
  ```

So, just get rid of this `grid` status and use only `row` and `onChange` to modify the data :)